.github/workflows/test-and-publish-nightly.yml

```diff
@@ -51,64 +51,101 @@ jobs:
             java-version: 11
             java-distribution: corretto
             container: "public.ecr.aws/async-profiler/asprof-builder-arm:latest"
+          - runson:
+              display: amazonlinux-2023 # Use dash instead of colon, because GHA artifacts cannot contain colons
+              name: ubuntu-latest
+            java-version: 11
+            java-distribution: corretto
+            container: amazonlinux:2023
+          - runson:
+              display: amazonlinux-2
+              name: ubuntu-latest
+            java-version: 11
+            java-distribution: corretto
+            # Make sure this is in sync with java-version/distribution for UX purposes
+            java-package: java-11-amazon-corretto
+            container: amazonlinux:2
     runs-on: ${{ matrix.runson.name }}
     container:
       image: ${{ matrix.container }}
       options: --privileged
     name: "test (${{ matrix.runson.display }}, ${{ matrix.java-distribution }} ${{ matrix.java-version }})"
     steps:
+      - name: Setup
+        run: |
+          if [[ "${{ matrix.container }}" == "amazonlinux"* ]]; then
+              yum update -y
+              yum install -y make gcc-c++ libstdc++-devel libstdc++-static util-linux tar git patchelf
```

Comment:
Does it make sense to prepare a container similar to `asprof-builder` so that you don't need to install all this stuff every time? In theory, you could also try to build node.js in this container that will be able to run standard github actions, but this is optional - I'm not sure how easy it will be,

Reply:
Node.js build is not too hard but will take a lot of time, having separate builder images sound better.

Reply:
Do we really want to use an outdated Node version? What if some any of the Actions we use (setup-java, checkout, download-artifacts, ...) breaks on older Node.js versions?

Reply:
I meant including new manually built version of Node.js in a container image

Reply:
If we want to have a new Docker image based on `amazonlinux:2` running a new Node.js version (probably `>20`) we'll need to change the environment a bit. The glibc version which comes out of the box with `amazonlinux:2` is not compatible for sure, thus we must change that at least.

Tweaking the environment this way means that we are also not going to test Async-Profiler on the same environment which is being used by `amazonlinux:2` users, which breaks a bit the motivation behind this PR. Am I missing something here @krk @apangin ?

Reply:
Node 20 (probably 22 as well) can be built with the libc in AL2, no update to the glibc version is necessary.